contrib/jemalloc/include/jemalloc/internal/jemalloc_internal_macros.h

```diff
@@ -115,10 +115,14 @@ JEMALLOC_DIAGNOSTIC_DISABLE_SPURIOUS
 #define	BOUND_PTR(ptr, size)	(ptr)
 #define	ROUND_SIZE(size)		(size)
 #else
-#define	BOUND_PTR(ptr, size)						\
-    ((ptr == NULL) ? NULL :						\
-	 cheri_andperm(cheri_setboundsexact((ptr), (size)),		\
-	     CHERI_PERMS_USERSPACE_DATA & ~CHERI_PERM_SW_VMEM))
+#define	BOUND_PTR(ptr, size) ({						\
+	typeof(ptr) _ptr = (ptr);					\
```

Comment:
Probably you could get by with just `void * _ptr` here if you wanted to?

Reply:
Then the macro won't be type-preserving like the intrinsics and the non-CHERI version